lib/utils/dom.js

```diff
@@ -2,6 +2,184 @@ var config = require('../config');
 
 var utils = {};
 
+/**
+ * Gets the previous link after `currentLink` within the containers array.
+ *
+ * @param      {HTMLElement}  containers    The containers array
+ * @param      {HTMLAnchorElement}  currentLink  The current link
+ * @returns    {HTMLAnchorElement} { Previous link }
+ */
+utils.getPreviousLink = function(containersArray, currentLink) {
+	if(!containersArray) return;
+
+	var links = utils.getLinks(containersArray),
+		currentIndex = links.indexOf(currentLink)|| 0,
+		previousIndex = currentIndex - 1;
+
+	previousIndex = wrapToEnd(previousIndex, links);
+	previousIndex = skipDisabledLinks(previousIndex, links, -1);
+
+	return links[previousIndex];
+};
+
+/**
+ * Gets the next link after `currentLink` within the containers array.
+ *
+ * @param      {HTMLElement}  containers    The containers array
+ * @param      {HTMLAnchorElement}  currentLink  The current link
+ * @returns    {HTMLAnchorElement} { Next link }
+ */
+utils.getNextLink = function(containersArray, currentLink) {
+	if(!containersArray) return;
+
+	var links = utils.getLinks(containersArray),
+		currentIndex = links.indexOf(currentLink) || 0;
+		nextIndex = currentIndex + 1;
+
+	nextIndex = wrapToEnd(nextIndex, links);
+	nextIndex = skipDisabledLinks(nextIndex, links, 1);
+
+	return links[nextIndex];
+};
+
+/**
+ * Gets the elements(Ids) which are controlled by elId.
+ * If 'elControlls' is null then there is nothing controlling the element passed in and so it is returned.
+ *
+ * @param      {HTMLElementId} elId Id of the element which controls other elements.
+ * @returns    {Array} { array of Ids }
+ */
+
+utils.getIdsOfControlledContainers = function(elId){
+	var elControls = document.getElementById(elId).getAttribute("aria-controls");
+	if(!elControls)
+		return null;
+	return elControls.split(" ");
+};
+
+/**
+ * Getst the element controlling the element passed in. If the controlling element is
+ *not accessible find the controlling element.
+ * @param      {HTMLElement} element you are trying to find the .
+ * @returns    {HTMLElementId} { Id }
+ */
+
+//passing in main-menu, splits out the same thing... why?
+utils.getControllingElementId = function(element){
+	if(!element.hasAttribute("aria-labelledby"))
+		return null;
+    var labelledBy = element.getAttribute("aria-labelledby");
+    var controllingEl = document.getElementById(labelledBy);
+
+    //If controlling El is disabled/hidden then find its parent (controlling) element
+	if(!isNavigable(controllingEl)){
+			var nextControllingEl = utils.closestWithClass(controllingEl, "menu");
+			if(nextControllingEl)
+				return utils.getControllingElementId(nextControllingEl);
+	}
+
+	return labelledBy;
+};
+
+/**
+* If the element passed in is accessible then true is retruned.
```

Comment:
"is accessible" should probably be "tabble" or "focussable" or something I think!